Client/Frontend/Browser/Playlist/Controllers/PlaylistMoveFolderView.swift

```diff
@@ -0,0 +1,269 @@
+                .padding(8.0)
+        }
+        .frame(width: 100.0, height: 60.0)
+        .onReceive(thumbnailLoader.$image) {
+            self.thumbnail = $0 ?? UIImage()
+        }
+        .onReceive(favIconLoader.$image) {
+            self.favIcon = $0 ?? UIImage()
+        }
+    }
+    
+    private class ImageLoader: ObservableObject {
+        @Published var image: UIImage?
+        
+        private let renderer = PlaylistThumbnailRenderer()
+        
+        init(thumbnail: PlaylistItem) {
+            guard let mediaSrc = thumbnail.mediaSrc,
+                  let assetUrl = URL(string: mediaSrc) else {
+                image = nil
+                return
+            }
+            
+            loadImage(url: assetUrl, isFavIcon: false)
+        }
+        
+        init(favIcon: PlaylistItem) {
+            guard let pageSrc = favIcon.pageSrc,
+                  let favIconUrl = URL(string: pageSrc) else {
+                image = nil
+                return
+            }
+            
+            loadImage(url: favIconUrl, isFavIcon: true)
+        }
+        
+        private func loadImage(url: URL, isFavIcon: Bool) {
+            renderer.loadThumbnail(assetUrl: isFavIcon ? nil : url,
+                                   favIconUrl: isFavIcon ? url : nil,
+                                   completion: { [weak self] image in
+                self?.image = image
+            })
+        }
+    }
+}
+
+private struct PlaylistFolderView: View {
+    let iconColor: Color
+    let titleColor: Color
+    let folder: PlaylistFolder?
+    @Binding var selectedFolder: PlaylistFolder?
+    
+    var body: some View {
+        HStack(spacing: 18.0) {
+            Image(systemName: "folder")
+                .renderingMode(.template)
+                .aspectRatio(contentMode: .fit)
+                .foregroundColor(iconColor)
+            
+            VStack(alignment: .leading) {
+                if let folder = folder {
+                    let itemCount = folder.playlistItems?.count ?? 0
+                    
+                    Text(folder.title ?? "")
+                        .font(.body)
+                        .foregroundColor(titleColor)
+                    Text("\(itemCount) \(itemCount == 1 ? Strings.PlaylistFolders.playlistFolderSubtitleItemSingleCount : Strings.PlaylistFolders.playlistFolderSubtitleItemCount)")
+                        .font(.footnote)
+                        .foregroundColor(titleColor)
+                } else {
+                    Spacer()
+                }
+            }
+            
+            if selectedFolder?.uuid == folder?.uuid {
+                Spacer()
+                Image(systemName: "checkmark")
+                    .renderingMode(.template)
+                    .aspectRatio(contentMode: .fit)
+                    .foregroundColor(.white)
+            }
+        }
+        .padding(EdgeInsets(top: 7.0, leading: 0.0, bottom: 7.0, trailing: 0.0))
```

Comment:
Replace this entirely with a `Label`:

```swift
Label {
  HStack {
    VStack(alignment: .leading) {
      Text(...) // name
      Text(...) // count
    }
    Spacer()
    if selectedFolder?.uuid == folder?.uuid {
      Image(sytemName: "checkmark").foregroundColor(.white)
    }
  }
} icon: {
  Image(systemName: "folder")
}
.padding(.vertical, 6)
```

1. It handles image alignment by default
2. You don't need to add Spacer's conditionally, their minWidth is 0
3. You don't need template rendering mode for SF Symbols

Reply:
If I do this, the image doesn't align center anymore. It will align top.
![image](https://user-images.githubusercontent.com/1530031/155237760-b083c57f-b21b-4e61-aeee-03fca44e7044.png)

Reply:
Ah crap good point, forgot `Label` only aligns to the first baseline

Reply:
In that case you need:

```swift
@ScaledMetric private var imageSize = 28

HStack(spacing: 10) {
  Image(...)
    .frame(width: imageSize)
  VStack { ... }
}
```

This will match iOS spacing properly even with dynamic fonts

Reply:
Fixed.